editor.planx.uk/src/@planx/components/Pay/Public/FeeBreakdown/utils.ts

```diff
@@ -5,24 +5,56 @@ import { FeeBreakdown, PassportFeeFields } from "./types";
 export const toNumber = (input: number | [number]) =>
   Array.isArray(input) ? input[0] : input;
 
+/**
+ * Convert a Passport value to an actual boolean
+ */
+const toBoolean = (val: ["true" | "false"]) => val[0] === "true";
+
+const filterByKeyPrefix = (data: Record<string, unknown>, prefix: string) =>
+  Object.fromEntries(
+    Object.entries(data).filter(([k, _v]) => k.startsWith(prefix))
+  );
+
+/**
+ * Iterate over exemptions or reductions to find matches, returning the granular keys
+ */
+  const getGranularKeys = (
+  data: Record<string, boolean> | undefined = {},
+  prefix: "application.fee.reduction" | "application.fee.exemption"
+) => {
+  const keys = Object.keys(data).filter((key) => data[key]);
+  const granularKeys = keys.map((key) => key.replace(prefix + ".", ""));
+  return granularKeys;
+};
+
 /**
  * A "reduction" is the sum of the difference between calculated and payable
- * This is not currently broken down further into component parts, or as exemptions or reductions
  */
-export const calculateReduction = (data: PassportFeeFields) =>
-  data["application.fee.calculated"]
-    ? data["application.fee.calculated"] - data["application.fee.payable"]
+export const calculateReduction = ({ amount }: PassportFeeFields) =>
+  amount["application.fee.calculated"]
+    ? amount["application.fee.calculated"] - amount["application.fee.payable"]
     : 0;
 
 /**
- * Transform Passport data to a FeeBreakdown shape
+ * Transform Passport data to a FeeBreakdown
  */
 export const toFeeBreakdown = (data: PassportFeeFields): FeeBreakdown => ({
-  applicationFee:
-    data["application.fee.calculated"] || data["application.fee.payable"],
-  total: data["application.fee.payable"],
-  vat: data["application.fee.payable.vat"],
-  reduction: calculateReduction(data),
+  amount: {
+    applicationFee:
+      data.amount["application.fee.calculated"] ||
+      data.amount["application.fee.payable"],
+    total: data.amount["application.fee.payable"],
+    vat: data.amount["application.fee.payable.vat"],
```

Comment:
I'll also pick up calculating this ourselves (and toggling the display in-component) as a follow on PR as discussed on the call this morning 👍